### PR TITLE
[DS-1355] Add new colorway design for Membership Calculator

### PR DESCRIPTION
**Pull request by aleevas**
Jira issue: [DS-1355](https://yusa.atlassian.net/browse/DS-1355)

### Steps to review

- log as admin
- create a new LP with LB
- add the Membership Calculator block
image

- save the node
- compare with a new design

### CODE REVIEW

File: openy_calc/assets/scss/main.scss [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+
+    &.btn-success {
+      background-color: unset;
+
+      .calc-summary {
+        color: var(--y-color-black, #000);
+        //a {
+        //
+        //}
+        //color: var(--y-color-black, #000);
```
Comment (rollki) [inline openy_calc/assets/scss/main.scss:330]: Could we remove these commented code?

File: openy_calc/assets/scss/main.scss [hunk 2]
```diff
@@ -0,0 +1,594 @@
...
+      flex-flow: row wrap;
+      width: 100%;
+      align-content: center;
+      justify-content: space-evenly;
+      align-items: flex-start;
+      padding: 36px;
+
+      &-col {
+        //min-height: 440px;
+
```
Comment (rollki) [inline openy_calc/assets/scss/main.scss:345]: Same here, just remove it

File: openy_calc/assets/scss/main.scss [hunk 3]
```diff
@@ -0,0 +1,594 @@
...
+              font-family: var(--ylb-font-family-verdana), serif;
+              font-weight: 400;
+              font-size: 18px;
+              line-height: 28px;
+            }
+          }
+        }
+
+        .card {
+          //min-height: 380px;
```
Comment (rollki) [inline openy_calc/assets/scss/main.scss:447]: Remove

File: openy_calc/assets/scss/main.scss [hunk 4]
```diff
@@ -0,0 +1,594 @@
...
+
+.price-month {
+  //margin-top: 20px;
+  //margin-bottom: 10px;
+  //font-size: $h2-font-size;
+  //font-family: $font-family-sans-serif;
+  //font-weight: 500;
+  //line-height: 1.1;
+  //color: inherit;
+}
```
Comment (rollki) [inline openy_calc/assets/scss/main.scss:594]: Lets just remove it

File: openy_calc/openy_calc.module [hunk 5]
```diff
@@ -48,3 +72,78 @@ function openy_calc_preprocess_fieldset(&$variables) {
...
+ */
+function openy_calc_preprocess_node__branch__calc_summary(&$variables) {
+  $node = $variables['node'];
+  if(!$node instanceof NodeInterface) {
+    return;
+  }
+//$branchHoursTodayHelper = \Drupal::service('lb_branch_hours_blocks.hours_today');
+//$branchHoursHelper = \Drupal::service('y_branch.hours_helper');
+//  $holiday_hours = $branchHoursHelper->getBranchHolidayHours($node);
+//  $branch_hours = $branchHoursHelper->getBranchHours($node);
```
Comment (rollki) [inline openy_calc/openy_calc.module:87]: Lets avoid commented code

File: openy_calc/src/Form/CalcBlockForm.php [hunk 6]
```diff
@@ -127,42 +127,56 @@ public function buildForm(array $form, FormStateInterface $form_state) {
...
     switch ($step) {
       case 1:
         // Membership type step.
-        $form['type'] = [
+        $form['membership_calc_body']['type'] = [
           '#element_variables' => $types,
           '#subtype' => 'membership_type_radio',
           '#type' => 'calc_radios',
-          '#title' => $this->t('Which option best describes the type of membership you need?'),
+          '#title' =>  '<h3>' . $this->t('Choose your membership type') . '</h3>',
```
Comment (rollki) [inline openy_calc/src/Form/CalcBlockForm.php:144]: Better to use Markup::
`        '#title' => Markup::create('<h3>' . $this->t('Choose your membership type') . '</h3>'),
`

File: openy_calc/src/Form/CalcBlockForm.php [hunk 7]
```diff
@@ -127,42 +127,56 @@ public function buildForm(array $form, FormStateInterface $form_state) {
...
-        $form['map'] = [
-          '#type' => 'openy_map',
-          '#element_variables' => $this->dataWrapper->getBranchPins(),
-        ];
         $locations = $this->dataWrapper->getLocations();
         $locations_options = [];
+        $locations_options[0] = $this->t('Choose location');
         foreach ($locations as $id => $location) {
           $locations_options[$id] = $location['title'];
         }
```
Comment (rollki) [inline openy_calc/src/Form/CalcBlockForm.php:157]: **Optional change:**
Single responsibility, so could be changed to something like
$locations_options = [0 => $this->t('Choose location')] + array_map(function ($location) {
  return $location['title'];
}, $locations);

File: openy_data_wrapper/src/DataWrapper.php [hunk 8]
```diff
@@ -217,6 +218,44 @@ public function getBranchPins($id = NULL) {
...
   }
 
+  /**
+   * Gets value for image field of template.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   An object of the Membership node.
+   *
+   * @return mixed|null
+   *   A value of the field.
```
Comment (rollki) [inline openy_data_wrapper/src/DataWrapper.php:228]: It should be array|null isn't it?
 * @return array|null
 *   A render array for the image or NULL if no image is found.

**Review by rollki — CHANGES_REQUESTED** [review]
Just a few comments from me, It looks good, thanks!

File: openy_calc/assets/scss/main.scss (hunk 1 above)
Comment (aleevas) [reply openy_calc/assets/scss/main.scss]: @rollki sorry, my bad.
Fixed

File: openy_calc/assets/scss/main.scss (hunk 2 above)
Comment (aleevas) [reply openy_calc/assets/scss/main.scss]: @rollki sorry, my bad.
Fixed

File: openy_calc/assets/scss/main.scss (hunk 3 above)
Comment (aleevas) [reply openy_calc/assets/scss/main.scss]: Fixed

File: openy_calc/assets/scss/main.scss (hunk 4 above)
Comment (aleevas) [reply openy_calc/assets/scss/main.scss]: Fixed

File: openy_calc/openy_calc.module (hunk 5 above)
Comment (aleevas) [reply openy_calc/openy_calc.module]: Fixed

File: openy_calc/src/Form/CalcBlockForm.php (hunk 6 above)
Comment (aleevas) [reply openy_calc/src/Form/CalcBlockForm.php]: Fixed

File: openy_calc/src/Form/CalcBlockForm.php (hunk 7 above)
Comment (aleevas) [reply openy_calc/src/Form/CalcBlockForm.php]: Fixed

File: openy_data_wrapper/src/DataWrapper.php (hunk 8 above)
Comment (aleevas) [reply openy_data_wrapper/src/DataWrapper.php]: fixed

File: openy_calc/openy_calc.module [hunk 9]
```diff
@@ -48,3 +71,74 @@ function openy_calc_preprocess_fieldset(&$variables) {
...
+      '#template' => '<h4>{{ branch_name }}</h4>',
+      '#context' => [
+        'branch_name' => $node->label(),
+      ],
+    ],
+    'branch_phone' => [
+      '#type' => 'inline_template',
+      '#template' => '<a href="tel:{{ branch_phone }}"><i class="fa fa-phone" aria-hidden="true"></i><div>{{ branch_phone }}</div></a>',
+      '#context' => [
+        'branch_phone' => $node?->get('field_location_phone')->value,
```
Comment (podarok) [inline openy_calc/openy_calc.module:96]: seems like we do expect $node could be NULL here which will make inline template on line 94 looks broken. We might need to have a condition, once $node is NULL - disaplay nothing or display placeholder @aleevas

File: openy_calc/openy_calc.module [hunk 10]
```diff
@@ -48,3 +71,74 @@ function openy_calc_preprocess_fieldset(&$variables) {
...
+      ],
+    ],
+    'branch_fax' => [
+      '#type' => 'inline_template',
+      '#template' => '{% if branch_fax  %}<a href="tel:{{ branch_fax }}"><i class="fa fa-phone" aria-hidden="true"></i><div>{{ branch_fax }}</div></a> {% endif %}',
+      '#context' => [
+        'branch_fax' => $node?->get('field_location_fax')->value,
```
Comment (podarok) [inline openy_calc/openy_calc.module:103]: same as https://github.com/open-y-subprojects/openy_custom/pull/76/files#r1694709572

File: openy_calc/openy_calc.module [hunk 11]
```diff
@@ -48,3 +71,74 @@ function openy_calc_preprocess_fieldset(&$variables) {
...
+      '#template' => '<div>{{ branch_address }}</div>',
+      '#context' => [
+        'branch_address' => _getAddress($node),
+      ],
+    ],
+    'branch_email' => [
+      '#type' => 'inline_template',
+      '#template' => '{% if branch_email %}<a href="mailto:{{ branch_email }}"><i class="fas fa-envelope"></i><div>{{ branch_email }}</div></a>{% endif %}',
+      '#context' => [
+        'branch_email' => $node?->get('field_location_email')->value,
```
Comment (podarok) [inline openy_calc/openy_calc.module:117]: same as https://github.com/open-y-subprojects/openy_custom/pull/76/files#r1694709572

File: openy_calc/openy_calc.info.yml [hunk 12]
```diff
@@ -9,3 +9,4 @@ dependencies:
   - openy_socrates:openy_socrates
   - openy_mappings:openy_mappings
   - openy_node_mbrshp:openy_node_mbrshp
+  - lb_branch_hours_blocks:lb_branch_hours_blocks
```
Comment (podarok) [inline openy_calc/openy_calc.info.yml:12]: We need to add hook_update_N with module enabling every time we add dependency to info file

File: openy_calc/openy_calc.libraries.yml [hunk 13]
```diff
@@ -1,16 +1,19 @@
...
     - core/jquery
     - core/drupal
     - core/drupalSettings
 
 submit:
-  version: 1.0
+  version: 1.1
   js:
-    js/openy_calc_submit.js: {}
+    assets/js/openy_calc_submit.js: {}
```
Comment (podarok) [inline openy_calc/openy_calc.libraries.yml:16]: let's add if it is minified or not

File: openy_calc/openy_calc.module (hunk 9 above)
Comment (david-storm) [reply openy_calc/openy_calc.module:96]: On line 80 we have condition $node, which it  instanceof NodeInterface so $node canot be NULL

File: openy_calc/openy_calc.module (hunk 10 above)
Comment (david-storm) [reply openy_calc/openy_calc.module:103]: On line 80 we have condition $node, which it  instanceof NodeInterface so $node canot be NULL

File: openy_calc/openy_calc.libraries.yml (hunk 13 above)
Comment (david-storm) [reply openy_calc/openy_calc.libraries.yml:16]: Added

File: openy_calc/openy_calc.info.yml (hunk 12 above)
Comment (david-storm) [reply openy_calc/openy_calc.info.yml:12]: Added hook_update_N

**Pull request merged**